components/script/dom/node.rs

```diff
@@ -1283,37 +1258,22 @@ impl Iterator for PrecedingNodeIterator {
     }
 }
 
-pub struct LastChildIterator {
-    current: Option<Root<Node>>,
-}
-
-impl Iterator for LastChildIterator {
-    type Item = Root<Node>;
-
-    fn next(&mut self) -> Option<Root<Node>> {
-        let current = match self.current.take() {
-            None => return None,
-            Some(current) => current,
-        };
-        self.current = current.GetLastChild();
-        Some(current)
-    }
-}
-
-pub struct AncestorIterator {
+struct SimpleNodeIterator<I>
```

Comment:
I would have called this just `NodeIterator`, but that's already a thing in dom/nodeiterator.rs